src/main/java/com/amazon/asksdk/cloudfoody/OpenTable.java

```diff
@@ -0,0 +1,42 @@
+package com.amazon.asksdk.cloudfoody;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.*;
+import java.net.*;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class OpenTable {
+
+    private static final String BASE_URL = "http://opentable.herokuapp.com/api/";
+
+    /*
+        The method establishes a connection and returns a url object
+     */
+    public static URLConnection createConnection(String urlString) throws IOException {
+        URL url = new URL(urlString);
+        return url.openConnection();
+    }
+
+
+    /*
+        The method takes a restaurant name as an arguement and returns back a hash with
+        all the info regarding that restaurant
+     */
+    public static HashMap getRestaurantByName(String name) throws IOException {
```

Comment:
Please return Restaurant instead of HashMap.

Reply:
In addition, please consider improving the error handling in this class. In addition to IOException, please think of handling service response errors (4xx, 5xx). If possible, try to throw specific errors, so we can handle various scenarios. E.g. if the service returns 404, throw some "ObjectNotFound" error (please check if service response errors are already defined in some library, e.g. java.net).